Framework/PythonInterface/plugins/algorithms/GSASIIRefineFitPeaks.py

```diff
@@ -15,974 +16,227 @@ class GSASIIRefineFitPeaks(PythonAlgorithm):
-        lattice_params = 7*[0.0]
-        parm_dict = {}
-        if "Pawley refinement" == method:
-            (gof_estimates, lattice_params) = self._run_rietveld_pawley_refinement(gs2, gs2_rd, True)
-        elif "Rietveld refinement" == method:
-            (gof_estimates, lattice_params) = self._run_rietveld_pawley_refinement(gs2, gs2_rd, False)
-        elif "Peak fitting" == method:
-            (gof_estimates, parm_dict) = self._run_peak_fit(gs2_rd, fit_inputs)
+        ws = self.getPropertyValue(self.PROP_INPUT_WORKSPACE)
+        if mtd[ws].getNumberHistograms > 1:
+            ws_index = self.getPropertyValue(self.PROP_WORKSPACE_INDEX)
+            spectrum = mantid.ExtractSpectra(InputWorkspace=ws, StartWorkspaceIndex=ws_index, EndWorkspaceIndex=ws_index)
+            mantid.DeleteWorkspace(Workspace=ws)
+            return spectrum
         else:
-            raise RuntimeError("Inconsistency found. Unknown refinement method: {0}".format(method))
+            return ws
 
-        return (gof_estimates, lattice_params, parm_dict)
-
-    def _run_rietveld_pawley_refinement(self, gs2, gs2_rd, do_pawley):
+    def _initialise_GSAS(self):
         """
-        Run Rietveld or Pawley refinement
-
-        @param do_pawley :: Select Pawley (True) or Rietveld (False)
-        @param gs2 :: the main GSAS-II object
-        @param gs2_rd :: the GSAS-II "rd" object with powder data in it. Phase information will be
-        connected to it
-
-        @return a tuple with 1) the two goodness-of-fit estimates, 2) lattice params (list with
-        7 values)
+        Initialise a GSAS project object with a spectrum and an instrument parameter file
+        :return: GSAS project object
         """
-        phase_data = self._load_prepare_phase_data(gs2, gs2_rd,
-                                                   self.getProperty(self.PROP_PHASE_INFO_FILE).value)
-
-        # Enable / tick on "Refine unit cell"
-        general_phase_data = phase_data['General']
-        general_phase_data['Cell'][0] = True
-        if do_pawley:
-            # Note from GSAS-II doc: "you probably should clear the Histogram scale factor refinement
-            # flag (found in Sample parameters for the powder data set) as it cannot be refined
-            # simultaneously with the Pawley reflection intensities"
-            gs2_rd.Sample['Scale'] = [1.0, False]
-
-            # Flag for Pawley intensity extraction (bool)
-            general_phase_data['doPawley'] = True
-            # maximum Q (as d-space) to use for Pawley extraction
-            general_phase_data['Pawley dmin'] = self.getProperty(self.PROP_PAWLEY_DMIN).value
-            # Restraint value for negative Pawley intensities
-            general_phase_data['Pawley neg wt'] = self.getProperty(self.PROP_PAWLEY_NEG_WEIGHT).value
+        gsas_path = self.getPropertyValue(self.PROP_PATH_TO_GSASII)
+        sys.path.append(gsas_path)
+        import GSASIIscriptable as GSASII
```

Comment:
This should be in a try/except and we need to report a very loud error to the log and the script window that GSAS is not installed and could not be found.